File: src/chipschallenge/blockreactions/ThinWallSeFrom.java
package chipschallenge.blockreactions;

import chipschallenge.Block;
import chipschallenge.BlockContainerFullException;
import chipschallenge.Move.Moves;

public class ThinWallSeFrom extends BlockReaction {

    private ThinWallSeFrom() {
    }
    private static ThinWallSeFrom mInstance = null;

    public static synchronized ThinWallSeFrom getInstance() {
        if (mInstance == null) {
            mInstance = new ThinWallSeFrom();
        }
        return mInstance;
    }

    public void react(Block moving, Block standing) throws BlockContainerFullException {
        // No reaction;
    }

    public boolean canMove(Block moving, Block standing) {
        return moving.getFacing() == Moves.UP || moving.getFacing() == Moves.LEFT;
    }
}

File: ChipsChallenge.Shared/Blockreactions/ThinWallSeFrom.cs
namespace ChipsChallenge.Shared.Blockreactions
{
    using Moves = Move.Moves;

    public class ThinWallSeFrom : NoSlipReaction
    {

        private ThinWallSeFrom()
        {
        }

        private static ThinWallSeFrom instance;

        public static ThinWallSeFrom Instance
        {
            get
            {
                lock (typeof(ThinWallSeFrom))
                {
                    if (instance == null)
                    {
                        instance = new ThinWallSeFrom();
                    }
                    return instance;
                }
            }
        }

        public override void React(Block moving, Block standing)
        {
        }

        public override bool canMove(Block moving, Block standing)
        {
            return moving.Facing == Moves.UP || moving.Facing == Moves.LEFT;
        }
    }
}